### PR TITLE
Improve code climate

### PR DESCRIPTION
**Pull request by smellsblue**
Add a code climate badge to the README and fix (hopefully) all of the issues they found

### CODE REVIEW

File: lib/gemstash/cli/setup.rb [hunk 1]
```diff
@@ -168,6 +156,14 @@ def with_new_config
       ensure
         gemstash_env.reset
       end
+
+      def check_for_exception(thing)
```
Comment (pcarranza) [inline lib/gemstash/cli/setup.rb]: Nitpicking here, what about s/check_for_exception/try/ ? (or probe)

**Comment by rjocoleman** [comment]
Let's get a `.codeclimate.yml` file in here too so it can be used with the [CLI](https://github.com/codeclimate/codeclimate) locally.

Potentially: 

```yaml
---
engines:
  bundler-audit:
    enabled: true
  duplication:
    enabled: true
    config:
      languages:
      - ruby
  fixme:
    enabled: true
  rubocop:
    enabled: true
ratings:
  paths:
  - Gemfile.lock
  - "**.rb"
exclude_paths:
- spec/
```

File: spec/gemstash/cli/setup_spec.rb [hunk 2]
```diff
@@ -56,4 +56,32 @@
...
+      expect(File.exist?(cli_options[:config_file])).to be_falsey
+      expect { Gemstash::CLI::Setup.new(cli).run }.to raise_error(Gemstash::CLI::Error, /The cache is not available/)
+      expect(File.exist?(cli_options[:config_file])).to be_falsey
+    end
+  end
+
+  context "with invalid database setup" do
+    let(:database) { double }
+
+    it "errors when cehcking the cache" do
```
Comment (pcarranza) [inline spec/gemstash/cli/setup_spec.rb]: s/cehcking/checking/

**Comment by pcarranza** [comment]
It looks good to me. 👍

File: spec/gemstash/cli/setup_spec.rb (hunk 2 above)
Comment (smellsblue) [inline spec/gemstash/cli/setup_spec.rb]: @rjocoleman fixed this in the auth branch, so I'll just merge it in as is and let that get fixed when the auth branch gets merged :-)

**Pull request merged**